File: backend/app/schemas/category.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class CategoryBase(BaseModel):
    name: str
    type: str  # "INCOME" or "EXPENSE"
    color: Optional[str] = None

class CategoryCreate(CategoryBase):
    pass

class CategoryUpdate(BaseModel):
    name: Optional[str] = None
    type: Optional[str] = None
    color: Optional[str] = None

class Category(CategoryBase):
    id: str
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/types/category.ts
export interface Category {
  id: string
  name: string
  type: 'INCOME' | 'EXPENSE'
  color?: string
  created_at: string
  updated_at: string
}

export interface CategoryCreate {
  name: string
  type: 'INCOME' | 'EXPENSE'
  color?: string
}

export interface CategoryUpdate {
  name?: string
  type?: 'INCOME' | 'EXPENSE'
  color?: string
}